packages/transaction-controller/src/TransactionController.ts

```diff
@@ -957,6 +971,53 @@ export class TransactionController extends BaseController<
     }
   }
 
+  /**
+   * Appends new sendFlowHistoryToAdd to the transaction with the given transactionID
+   * if transaction is unapproved. Returns the updated transactionMeta.
+   *
+   * @param transactionID - The ID of the transaction to update.
+   * @param currentSendFlowHistoryLength - The length of the current sendFlowHistory array.
+   * @param sendFlowHistoryToAdd - The sendFlowHistory entries to add.
+   * @returns The updated transactionMeta.
+   */
+  updateTransactionSendFlowHistory(
+    transactionID: string,
+    currentSendFlowHistoryLength: number,
+    sendFlowHistoryToAdd: SendFlowHistoryEntry[],
+  ): TransactionMeta {
+    if (this.isSendFlowHistoryDisabled) {
+      throw new Error(
+        'TransactionsController: Send flow history is disabled for this instance of the controller.',
+      );
+    }
+
+    const transactionMeta = this.getTransaction(transactionID);
+
+    if (!transactionMeta) {
+      throw new Error(
+        `TransactionsController: updateTransactionSendFlowHistory called but transactionMeta not found.`,
```

Comment:
```suggestion
        `Cannot update send flow history as no transaction metadata found`,
```

Reply:
I appreciate you're matching the exception messages from the extension, but is it safe to assume no existing logic is relying on the message string itself?